compiler/src/dotty/tools/dotc/interactive/Interactive.scala

```diff
@@ -166,31 +166,72 @@ object Interactive {
   private def computeCompletions(pos: SourcePosition, path: List[Tree])(implicit ctx: Context): (Int, List[Symbol]) = {
     val completions = Scopes.newScope.openForMutations
 
-    val (completionPos, prefix, termOnly, typeOnly) = path match {
+    /**
+     * Extract basic info about completion location and the kind of symbols to include.
+     *
+     * @param path     The path to the position where completion happens
+     * @param inImport If set, indicates that this is the completion of an import node. When
+     *                 completing imports, both types and terms are always included.
+     * @return The point where to insert completion, whether terms should be included in results,
+     *         whether types should be included, and whether we're completing an import.
+     */
+    def completionInfo(path: List[Tree], inImport: Boolean): (Int, String, Boolean, Boolean, Boolean) = path match {
       case (ref: RefTree) :: _ =>
         if (ref.name == nme.ERROR)
-          (ref.pos.point, "", false, false)
+          (ref.pos.point, "", false, false, inImport)
         else
           (ref.pos.point,
            ref.name.toString.take(pos.pos.point - ref.pos.point),
-           ref.name.isTermName,
-           ref.name.isTypeName)
+           !inImport && ref.name.isTermName, // Types and terms are always accepted in imports
+           !inImport && ref.name.isTypeName,
+           inImport)
       case _ =>
-        (0, "", false, false)
+        (0, "", false, false, false)
+    }
+
+    val (completionPos, prefix, termOnly, typeOnly, inImport) = path match {
+      case (Thicket(name :: _ :: Nil)) :: (imp: Import) :: _ =>
+        if (name.pos.contains(pos.pos))
+          completionInfo(name.asInstanceOf[tpd.Tree] :: Nil, /* inImport = */ true)
+        else completionInfo(path, /* inImport = */ true)
+
+      case (imp: Import) :: _ =>
+        imp.selectors.find(_.pos.contains(pos.pos)) match {
+          case None      => (imp.expr.pos.point, "", false, false, true)
+          case Some(sel) => completionInfo(sel.asInstanceOf[tpd.Tree] :: Nil, /* inImport = */ true)
+        }
+
+      case other =>
+        completionInfo(other, /* inImport = */ false)
     }
 
     /** Include in completion sets only symbols that
      *   1. start with given name prefix, and
      *   2. do not contain '$' except in prefix where it is explicitly written by user, and
-     *   3. have same term/type kind as name prefix given so far
+     *   3. are not a primary constructor,
+     *   4. have an existing source symbol,
+     *   5. are the module class in case of packages,
+     *   6. are not Java module classes when completing imports (to avoid duplicate results).
```

Comment:
Don't you also get duplicates if there's both a Scala class and object with the same name ?

Reply:
Yes, but only if there explicitly exists a class and an object, so I personally find it less disturbing

https://github.com/lampepfl/dotty/blob/eba580ccf17c37cde6562beefad5e7f0b0345029/language-server/test/dotty/tools/languageserver/CompletionTest.scala#L83-L91

I'm happy to just remove this commit if you think it's not worth it. Or we could bypass the problem by removing all duplicate names from the completion lists. Or keep the duplicates.

Reply:
> Or we could bypass the problem by removing all duplicate names from the completion lists. 

Yeah I would just do that, if there's both a class and an object, then just show the symbol for class in the completion

Reply:
Also ideally in term position we should just show the objects, and in type position the class.